### ISSUE TITLE
mypy wants `Future[<nothing>]`, thinks it's incompatible with `Future[str]`

### ISSUE THREAD

**Issue by gjcarneiro** [python/mypy#8546]
Apologies if this is a typeshed issue.  It happened when I upgraded mypy, so filing with mypy first.

* Are you reporting a bug, or opening a feature request?

bug

* Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.

```python
import asyncio
from typing import List

async def main() -> None:
    missing: List[asyncio.Future[str]] = []
    await asyncio.wait(missing)

asyncio.run(main())
```

* What is the actual behavior/output?

```
test.py:7: error: Argument 1 to "wait" has incompatible type "List[Future[str]]"; expected "Iterable[Union[Future[<nothing>], Generator[Any, None, <nothing>], Awaitable[<nothing>]]]"
Found 1 error in 1 file (checked 1 source file)
```

* What is the behavior/output you expect?

I expected no error.  Just because a future has a value doesn't mean it's incompatible with asyncio.wait()

* What are the versions of mypy and Python you are using?
  Do you see the same issue after installing mypy from Git master?

Problem is with mypy 0.750.  I upgraded from 0.641, which is when it started happening, version 0.641 was fine.  Didn't check master.

cpython 3.7.3

* What are the mypy flags you are using? (For example --strict-optional)

No flags.

**Comment by JukkaL** [comment]
This doesn't generate an error on the latest GitHub master.

If you can reproduce the error on GitHub master, we can reopen this issue.
